packages/flutter/test/material/tabs_utils.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+RenderParagraph getTabText(WidgetTester tester, String text) {
+  return tester.renderObject<RenderParagraph>(find.text(text));
+}
+
+TabController getTabController({required int length, required TickerProvider vsync, int initialIndex = 0, Duration? animationDuration}) {
+  final TabController result = TabController(length: length, vsync: vsync, initialIndex: initialIndex, animationDuration: animationDuration);
+  addTearDown(result.dispose);
+  return result;
+}
+
+typedef TabControllerFrameBuilder = Widget Function(BuildContext context, TabController controller);
+
+class TabStateMarker extends StatefulWidget {
```

Comment:
Since we're prepping this to be used across multiple test files, can you add some comments explaining what these classes can be used for?

Reply:
Great idea, i added some comments.